src/common/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) 2017-2024 Smart code 203358507
+
+import React, { useState, useEffect, useCallback, DetailedHTMLProps, HTMLAttributes } from 'react';
+import classNames from 'classnames';
+import styles from './Checkbox.less';
+import Icon from '@stremio/stremio-icons/react';
+
+type Props = {
+    disabled?: boolean;
+    value?: boolean;
+    className?: string;
+    onChange?: (checked: boolean) => void;
+    ariaLabel?: string;
+    error?: string;
+};
+
+const Checkbox = ({ disabled, value, className, onChange, ariaLabel, error }: Props) => {
+    const [isChecked, setIsChecked] = useState(false);
+    const [isError, setIsError] = useState(false);
+    const [isDisabled, setIsDisabled] = useState(disabled);
+
+    const handleChangeCheckbox = useCallback(() => {
+        if (disabled) {
+            return;
+        }
+
+        setIsChecked(!isChecked);
+        onChange && onChange(!isChecked);
+    }, [disabled]);
+
+    const handleEnterPress = useCallback((event: DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>) => {
+        if ((event.key === 'Enter' || event.key === ' ') && !disabled) {
+            setIsChecked(!isChecked);
```

Comment:
In each functions that does this, instead of `setIsChecked(!isChecked);` use `setIsChecked((isChecked) => !isChecked);`
And add a useEffect instead of `onChange && onChange(!isChecked);` 
```
useEffect(() => {
    onChange && onChange(isChecked);
}, [isChecked, onChange]);
```